### PR TITLE
Google chat via webhook fix outputs

### PR DESCRIPTION
**Pull request by edibleShell**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
The Google Chat API was recently updated with new outputs. This PR includes changes to the integration commands to support the new outputs from the API. 

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py [hunk 1]
```diff
@@ -121,17 +121,10 @@ def send_google_chat_message_command(client: Client, message: str, threadName: O
     """
     res = client.send_google_chat_message(message=message, threadName=threadName)
     result = {
+        'Name': res.get('name'),
         'Message': res.get('text'),
-        'SpaceName': res.get('space').get('name'),
```
Comment (tcarmeli1) [inline Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py:125]: Why all the outputs below are deleted?

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py [hunk 2]
```diff
@@ -159,16 +152,9 @@ def send_google_chat_custom_card_command(client: Client, blocks: str, threadName
     """
     res = client.send_google_chat_custom_card(blocks=blocks, threadName=threadName)
     result = {
-        'SpaceName': res.get('space').get('name'),
```
Comment (tcarmeli1) [inline Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py:162]: Why all the outputs below are deleted?

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml [hunk 3]
```diff
@@ -45,28 +45,14 @@ script:
     description: Send a message to Google Chat Space via Incoming Webhook.
     name: send-google-chat-message
     outputs:
-    - contextPath: GoogleChatWebhook.Message.SpaceType
```
Comment (tcarmeli1) [inline Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml:48]: Same here.

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml [hunk 4]
```diff
@@ -76,33 +62,17 @@ script:
     description: Send a customizable card to Google Chat Space via Incoming Webhook.
     name: send-google-chat-custom-card
     outputs:
-    - contextPath: GoogleChatWebhook.CustomCard.Cards
```
Comment (tcarmeli1) [inline Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml:79]: Same here.

**Review by tcarmeli1 — CHANGES_REQUESTED** [review]
Hi @edibleShell ,
Thank you for your contribution!
Good work :)
Please see my comments.
Thank you again.

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml (hunk 4 above)
Comment (edibleShell) [reply Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml:79]: The GoogleChat API changed its outputs and no longer includes all of those values. We now receive the following output when interacting with the API. This is why I modified all of the outputs in this PR. 

`{
    "name": "spaces/AAAA5weu7HQ/messages/h229LcOBQTA.h229LcOBQTA",
    "text": "Hello from the python test client",
    "thread": {
        "name": "spaces/AAAA5weu7HQ/threads/h229LcOBQTA"
    },
    "space": {
        "name": "spaces/AAAA5weu7HQ"
    }
}`

Comment (tcarmeli1) [reply Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml:79]: if so, lets add a breaking changes as you can see here https://xsoar.pan.dev/docs/documentation/release-notes#breaking-changes-version
In addition, please update the release notes to be more informative about the cause of the change (deleted outputs) and list that we also removed some outputs (list them).
Thank you!

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py [hunk 5]
```diff
@@ -121,17 +121,10 @@ def send_google_chat_message_command(client: Client, message: str, threadName: O
...
-        'SpaceDisplayName': res.get('space').get('displayName'),
-        'SpaceType': res.get('space').get('type'),
-        'CreatedTime': res.get('createTime'),
-        'ThreadReply': res.get('threadReply', False),
         'ThreadName': res.get('thread').get('name'),
-        'Name': res.get('name'),
-        'SenderDisplayName': res.get('sender').get('displayName'),
-        'SenderName': res.get('sender').get('name'),
-        'SenderType': res.get('sender').get('type')
+        'Space': res.get('space').get('name')
```
Comment (tcarmeli1) [inline Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py:127]: The previous version called it 'SpaceName' lets leave it the same

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py [hunk 6]
```diff
@@ -159,16 +152,9 @@ def send_google_chat_custom_card_command(client: Client, blocks: str, threadName
...
-        'SpaceType': res.get('space').get('type'),
-        'CreatedTime': res.get('createTime'),
-        'ThreadReply': res.get('threadReply', False),
-        'ThreadName': res.get('thread').get('name'),
         'Name': res.get('name'),
-        'SenderDisplayName': res.get('sender').get('displayName'),
-        'SenderName': res.get('sender').get('name'),
-        'SenderType': res.get('sender').get('type')
+        'ThreadName': res.get('thread').get('name'),
+        'Space': res.get('space').get('name')
```
Comment (tcarmeli1) [inline Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py:157]: The previous version called it 'SpaceName' lets leave it the same

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml [hunk 7]
```diff
@@ -45,28 +45,14 @@ script:
...
-      description: Google Chat message creation time.
+      description: Google Chat message name.
+    - contextPath: GoogleChatWebhook.Message.Message
+      description: Google Chat message that was sent.
     - contextPath: GoogleChatWebhook.Message.ThreadName
-      description: Google Chat thread name.
-    - contextPath: GoogleChatWebhook.Message.SenderDisplayName
-      description: Google Chat message sender display name.
+      description: Google Chat message Thread name.
+    - contextPath: GoogleChatWebhook.Message.Space
```
Comment (tcarmeli1) [inline Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml:54]: please use SpaceName

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py (hunk 6 above)
Comment (edibleShell) [reply Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py]: Done

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py (hunk 5 above)
Comment (edibleShell) [reply Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.py]: Done

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml (hunk 4 above)
Comment (edibleShell) [reply Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml:79]: Done

Comment (tcarmeli1) [reply Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml:79]: please do not remove the GoogleChatWebhook.CustomCard.Cards output as we do receive it in the response
thanks!

File: Packs/GoogleChatViaWebhook/ReleaseNotes/1_0_1.json [hunk 8]
```diff
@@ -0,0 +1,4 @@
+{
+    "breakingChanges": true,
+    "breakingChangesNotes": "Deleted (SpaceDisplayName, SpaceType, CreatedTime, ThreadReply, SenderDisplayName, SenderName, SenderType) to align with Google Chat Webhook API outputs. These outputs were also removed from Context Data outputs in the integration settings."
```
Comment (tcarmeli1) [inline Packs/GoogleChatViaWebhook/ReleaseNotes/1_0_1.json:3]: please change it to:
Deleted SpaceDisplayName, SpaceType, CreatedTime, ThreadReply, SenderDisplayName, SenderName, SenderType from the outputs to align with the Google Chat Webhook API response change which does not support it.

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml (hunk 4 above)
Comment (tcarmeli1) [inline Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml:79]: please do not remove the GoogleChatWebhook.CustomCard.Cards output as we do receive it in the response (it is under cardsV2 in the response and then being handled in line 162)
thanks!

**Review by tcarmeli1 — CHANGES_REQUESTED** [review]
Hi @edibleShell ,
Great work!
Thank you for your response.
I added two additional comments.
I would be happy if we can conduct a Demo meeting - please let me know when are you available (you can also send me a Demo video - if so please show me the commands and the raw response from the api so i can see the change in the api response)
Thank you so much!

File: Packs/GoogleChatViaWebhook/ReleaseNotes/1_0_1.json (hunk 8 above)
Comment (edibleShell) [reply Packs/GoogleChatViaWebhook/ReleaseNotes/1_0_1.json]: Done

File: Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml (hunk 4 above)
Comment (edibleShell) [reply Packs/GoogleChatViaWebhook/Integrations/GoogleChatViaWebhook/GoogleChatViaWebhook.yml:79]: Done

**Review by tcarmeli1 — CHANGES_REQUESTED** [review]
Hi @edibleShell ,
Thank you for your great demo!
Could you please show me an execution with raw_response=false so i can see the table on the war room?
Also, if you have any documentation that shows the Google Chat API has changed the response, or any confirmation that some fields are no longer being returned (since I checked their documentation and couldn't find it here: https://developers.google.com/workspace/chat/api/reference/rest/v1/spaces.messages/create or in their release notes), I would appreciate it.
I just want to ensure we’re not missing something, or if adding a parameter could help return the original response, as these is breaking changes and we want to prevent them as much as possible.
Thank you very much!

**Pull request closed without merging**